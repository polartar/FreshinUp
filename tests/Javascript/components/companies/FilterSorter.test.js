import { mount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import Component from '~/components/companies/FilterSorter.vue'

describe('Companies FilterSorter Component', () => {
  let localVue

  beforeEach(() => {
    const vue = createLocalVue({ validation: true })
    localVue = vue.localVue
  })

  describe('Mount', () => {
    test('snapshot', () => {
      const wrapper = mount(Component, {
        localVue
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    test('run() emits runFilter', () => {
      const wrapper = mount(Component, {
        localVue
      })
      wrapper.vm.run({ term: 'test' })
      expect(wrapper.emitted().runFilter).toBeTruthy()
      expect(wrapper.emitted().runFilter).toHaveLength(1)
      expect(wrapper.emitted().runFilter[0]).toEqual([{ 'filter[name]': 'test', 'filter[status]': null, 'filter[users_id]': null, 'sort': undefined }])
    })

    test('selectOwner() emits runFilter and change value to company', () => {
      const wrapper = mount(Component, {
        localVue
      })
      wrapper.vm.selectOwner({ id: 1 })
      expect(wrapper.vm.owner).toBe(1)
      expect(wrapper.emitted().runFilter).toBeTruthy()
      expect(wrapper.emitted().runFilter).toHaveLength(1)
      expect(wrapper.emitted().runFilter[0]).toEqual([{ 'filter[name]': '', 'filter[status]': null, 'filter[users_id]': 1, 'sort': 'created_at' }])
    })
  })
})
