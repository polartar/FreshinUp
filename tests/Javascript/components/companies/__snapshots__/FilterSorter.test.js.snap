// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies FilterSorter Component Mount snapshot 1`] = `
<div
  autocomplete_url="/api/companies"
  class="v-card v-sheet theme--light"
  filters="[object Object]"
  style="background-color: rgb(248, 248, 248); border-color: #f8f8f8;"
>
  <form
    class="v-form"
    novalidate="novalidate"
  >
    <div
      class="container filter-sorter-container fluid"
    >
       
      <div
        class="layout filter-sorter-main-layout row wrap"
      >
        <div
          class="flex shrink"
        >
          <div
            class="v-card__text"
          >
            Filter:
          </div>
        </div>
         
        <div
          class="flex grow"
        >
          <div
            class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder theme--light"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-text-field__slot"
                >
                  <input
                    placeholder="Start typing to Search"
                    type="text"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex shrink"
        >
          <div
            class="v-input filter-sorter-sortable-select v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-select theme--light"
            role="combobox"
          >
            <div
              class="v-input__control"
            >
              <div
                class="v-input__slot"
              >
                <div
                  class="v-select__slot"
                >
                  <label
                    aria-hidden="true"
                    class="v-label theme--light"
                    style="left: 0px; position: absolute;"
                  >
                    Sort by
                  </label>
                  <div
                    class="v-select__selections"
                  >
                    <input
                      aria-label="Sort by"
                      aria-readonly="false"
                      autocomplete="on"
                      data-vv-name="sort"
                      readonly="readonly"
                      type="text"
                    />
                  </div>
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon material-icons theme--light"
                      >
                        arrow_drop_down
                      </i>
                    </div>
                  </div>
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light "
                    style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  >
                    <div
                      class="v-select-list v-card theme--light"
                    >
                      <div
                        class="v-list theme--light"
                        role="list"
                      >
                        <div
                          role="listitem"
                        >
                          <div
                            class="v-list__tile theme--light"
                          >
                            <div
                              class="v-list__tile__content"
                            >
                              <div
                                class="v-list__tile__title"
                              >
                                No data available
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <div
                    class="v-messages__wrapper"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex shrink"
        >
          <button
            class="v-btn v-btn--large theme--light ma-0 "
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            More Filters
          
            </div>
          </button>
        </div>
         
      </div>
       
      <!---->
       
      <div
        class="layout filter-sorter-expanded-layout align-center justify-center row wrap fill-height"
        style="display: none;"
      >
        <div
          class="flex"
        >
          <div
            class="v-card__text"
          >
            <div
              class="layout row wrap"
            >
              <div
                class="flex lg4"
              >
                <div
                  class="layout row"
                >
                  <div
                    class="flex shrink"
                  >
                    <div
                      class="v-card__text"
                    >
                      Company Type
                    </div>
                  </div>
                   
                  <div
                    class="flex"
                  >
                    <div
                      class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder v-select v-input--hide-details theme--light"
                      role="combobox"
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot"
                        >
                          <div
                            class="v-select__slot"
                          >
                            <div
                              class="v-select__selections"
                            >
                              <input
                                aria-readonly="false"
                                autocomplete="on"
                                placeholder="All Types"
                                readonly="readonly"
                                type="text"
                              />
                            </div>
                            <div
                              class="v-input__append-inner"
                            >
                              <div
                                class="v-input__icon v-input__icon--append"
                              >
                                <i
                                  aria-hidden="true"
                                  class="v-icon material-icons theme--light"
                                >
                                  arrow_drop_down
                                </i>
                              </div>
                            </div>
                          </div>
                          <div
                            class="v-menu"
                          >
                            <div
                              class="v-menu__content theme--light "
                              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                            >
                              <div
                                class="v-select-list v-card theme--light"
                              >
                                <div
                                  class="v-list theme--light"
                                  role="list"
                                >
                                  <div
                                    role="listitem"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <div
                                        class="v-list__tile__content"
                                      >
                                        <div
                                          class="v-list__tile__title"
                                        >
                                          No data available
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="flex lg4"
              >
                <div
                  class="layout row"
                >
                  <div
                    class="flex shrink"
                  >
                    <div
                      class="v-card__text"
                    >
                      Status
                    </div>
                  </div>
                   
                  <div
                    class="flex"
                  >
                    <div
                      class="v-input v-text-field v-text-field--single-line v-text-field--solo v-text-field--solo-flat v-text-field--enclosed v-text-field--placeholder v-select v-input--hide-details theme--light"
                      role="combobox"
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot"
                        >
                          <div
                            class="v-select__slot"
                          >
                            <div
                              class="v-select__selections"
                            >
                              <input
                                aria-readonly="false"
                                autocomplete="on"
                                placeholder="All"
                                readonly="readonly"
                                type="text"
                              />
                            </div>
                            <div
                              class="v-input__append-inner"
                            >
                              <div
                                class="v-input__icon v-input__icon--append"
                              >
                                <i
                                  aria-hidden="true"
                                  class="v-icon material-icons theme--light"
                                >
                                  arrow_drop_down
                                </i>
                              </div>
                            </div>
                          </div>
                          <div
                            class="v-menu"
                          >
                            <div
                              class="v-menu__content theme--light "
                              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                            >
                              <div
                                class="v-select-list v-card theme--light"
                              >
                                <div
                                  class="v-list theme--light"
                                  role="list"
                                >
                                  <div
                                    role="listitem"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <div
                                        class="v-list__tile__content"
                                      >
                                        <div
                                          class="v-list__tile__title"
                                        >
                                          No data available
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
               
              <div
                class="flex lg4"
              >
                <div
                  class="layout row"
                >
                  <div
                    class="flex shrink"
                  >
                    <div
                      class="v-card__text"
                    >
                      Owner
                    </div>
                  </div>
                   
                  <div
                    class="flex"
                  >
                    <div
                      background-color="white"
                      class="v-input mt-0 pt-0 v-text-field v-text-field--placeholder v-select v-autocomplete v-input--hide-details theme--light"
                      height="48"
                    >
                      <div
                        class="v-input__control"
                      >
                        <div
                          class="v-input__slot white"
                          style="height: 48px;"
                        >
                          <div
                            class="v-select__slot"
                          >
                            <input
                              aria-label=""
                              autocomplete="off"
                              placeholder="All"
                              role="combobox"
                              type="text"
                            />
                            <div
                              class="v-input__append-inner"
                            >
                              <div
                                class="v-input__icon v-input__icon--"
                              >
                                <i
                                  aria-hidden="true"
                                  class="v-icon v-icon--link material-icons theme--light"
                                />
                              </div>
                            </div>
                            <div
                              class="v-input__append-inner"
                            >
                              <div
                                class="v-input__icon v-input__icon--append"
                              >
                                <i
                                  aria-hidden="true"
                                  class="v-icon material-icons theme--light"
                                >
                                  arrow_drop_down
                                </i>
                              </div>
                            </div>
                          </div>
                          <div
                            class="v-menu"
                          >
                            <div
                              class="v-menu__content theme--light v-autocomplete__content"
                              style="max-height: 300px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                            >
                              <div
                                class="v-select-list v-card theme--light"
                              >
                                <div
                                  class="v-list theme--light"
                                  role="list"
                                >
                                  <div
                                    role="listitem"
                                  >
                                    <div
                                      class="v-list__tile theme--light"
                                    >
                                      <div
                                        class="v-list__tile__content"
                                      >
                                        <div
                                          class="v-list__tile__title"
                                        >
                                          No data available
                                        </div>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="flex text-no-wrap"
        >
          <button
            class="v-btn v-btn--large theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            Clear Filters
          
            </div>
          </button>
        </div>
      </div>
    </div>
  </form>
</div>
`;
